File: back-end/models/Console.go
package models

import (
	"time"

	"github.com/go-playground/validator/v10"
)

type Console struct {
	IdConsole      uint         `gorm:"primaryKey" json:"id_console"`
	NameConsole    string       `json:"name_console"`
	ManufacturerID uint         `json:"manufacturer_id"`
	Manufacturer   Manufacturer `gorm:"foreignKey:ManufacturerID" json:"-"`
	ReleaseDate    string       `json:"release_date"`
	IsActive       bool         `json:"is_active"`
	CreatedAt      time.Time    `json:"created_at"`
	UpdatedAt      time.Time    `json:"updated_at"`
}

func (console *Console) Validate() error {
	validate := validator.New()
	return validate.Struct(console)
}

File: front-end/web/src/app/models/console.ts
import { Manufacturer } from "./manufacturer";

export class Console {
  id_console: number;
  name_console: string;
  manufacturer_id: number;
  manufacturer?: Manufacturer;
  release_date: string;
  is_active: boolean;
  created_at: Date;
  updated_at: Date;

  constructor() {
    this.id_console = 0;
    this.name_console = '';
    this.manufacturer_id = 0;
    this.release_date = '';
    this.is_active = true;
    this.manufacturer = new Manufacturer();
    this.created_at = new Date();
    this.updated_at = new Date();
  }
}
